File: Linkedlist/js/getNodeValue.js
class Node{
    constructor(val, next= null){
        this.val = val
        this.next = next
    }
}


function getNodeValue(head, index){
    let curr = head
    let count = 0

    while(curr !== null){
        if(index === count) return curr.val
        count ++
        curr = curr.next
    }
    return null
}

const getNodeValue = (head, index) => {
  if (head === null) return null;
  if (index === 0) return head.val;
  return getNodeValue(head.next, index - 1);
};

File: Linkedlist/python/getNodeValue.py
from __future__ import annotations
from dataclasses import dataclass
from typing import Optional


@dataclass
class Node:
    val: int
    next: Optional["Node"] = None


def get_node_value(head: Optional[Node], index: int) -> Optional[int]:
    current = head
    position = 0
    while current is not None:
        if position == index:
            return current.val
        position += 1
        current = current.next
    return None
